lib/block_view.go

```diff
@@ -1599,9 +1599,24 @@ func (bav *UtxoView) _checkDerivedKeySpendingLimit(
 		}
 	case TxnTypeDAOCoinLimitOrder:
 		txnMeta := txn.TxnMeta.(*DAOCoinLimitOrderMetadata)
+		var buyingCoinPublicKey []byte
+		var sellingCoinPublicKey []byte
+		if txnMeta.CancelOrderID != nil {
+			orderEntry, err := bav._getDAOCoinLimitOrderEntry(txnMeta.CancelOrderID)
+			if err != nil || orderEntry == nil {
+				return utxoOpsForTxn, errors.Wrapf(
+					RuleErrorDerivedKeyInvalidDAOCoinLimitOrderOrderID,
+					"_checkDerivedKeySpendingLimit: Invalid DAO coin limit order ID %v",
+					txnMeta.CancelOrderID)
+			}
+			buyingCoinPublicKey = bav.GetPublicKeyForPKID(orderEntry.BuyingDAOCoinCreatorPKID)
```

Comment:
There's a small edge case here where someone provides a CancelOrderID and a BuyingDAOCoinCreatorPublicKey and/or SellingDAOCoinCreatorPublicKey in their metadata. We ignore the Buying and Selling DAOCoinCreatorPublicKeys in that case. IMO that's fine. One could argue that that's confusing for the user, but maybe, maybe not that's true. Alternatively, we could throw a new error like cannot specify buying public key when cancelling an order. And the same for selling public key.

Reply:
Good point, throwing an error clearly signals to the user that the buying and selling coin public keys will never be used. However, this should defer to whatever the behavior is in [_connectDAOCoinLimitOrder](https://github.com/deso-protocol/core/blob/main/lib/block_view_dao_coin_limit_order.go#L282). There, we quietly ignore the buying and selling coin fields if the CancelOrderID field is provided.

IMO, any transaction that goes through when signed by the user's private key should go through when signed by a valid derived key (and vice versa). Similarly, any transaction that's rejected when signed by one should be rejected when signed by the other. Inconsistency here would be a bigger point of confusion.

Adding the check and throwing the error in _connectDAOCoinLimitOrder would keep the two paths consistent, but do we really want to introduce new behavior there?